### PR TITLE
start bookstore project v1.0

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/java/com/parkhomovsky/bookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (ConfusedSatlan) [inline src/main/java/com/parkhomovsky/bookstore/model/Book.java]: add name for each column for db

File: src/main/java/com/parkhomovsky/bookstore/service/implementation/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        this.bookRepository = bookRepository;
+    }
+
+    @Override
+    public Book save(Book book) {
+        return bookRepository.save(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
```
Comment (ConfusedSatlan) [inline src/main/java/com/parkhomovsky/bookstore/service/implementation/BookServiceImpl.java]: in service we use create, get naming for repository we use save and find naming

**Review by ConfusedSatlan — CHANGES_REQUESTED** [review]
Well done, all correct. But need some minor improvements

File: src/main/java/com/parkhomovsky/bookstore/service/implementation/BookServiceImpl.java (hunk 2 above)
Comment (ConfusedSatlan) [reply src/main/java/com/parkhomovsky/bookstore/service/implementation/BookServiceImpl.java]: in this case getAll()

File: src/main/java/com/parkhomovsky/bookstore/model/Book.java (hunk 1 above)
Comment (CodeWebWeaver) [reply src/main/java/com/parkhomovsky/bookstore/model/Book.java]: Why?
is there any reason to name "price" as "price"?

File: src/main/java/com/parkhomovsky/bookstore/service/implementation/BookServiceImpl.java (hunk 2 above)
Comment (Skarbey) [reply src/main/java/com/parkhomovsky/bookstore/service/implementation/BookServiceImpl.java:25]: Can you rebuild your sentence? I don`t understand

File: src/main/java/com/parkhomovsky/bookstore/model/Book.java (hunk 1 above)
Comment (ConfusedSatlan) [reply src/main/java/com/parkhomovsky/bookstore/model/Book.java]: the task says to set the name of each column

File: src/main/java/com/parkhomovsky/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private String title;
+    @Column(name = "Author", nullable = false)
+    private String author;
+    @Column(name = "ISBN",nullable = false, unique = true)
+    private String isbn;
+    @Column(name = "Price", nullable = false)
+    private BigDecimal price;
+    @Column(name = "Description")
+    private String description;
+    @Column(name = "Image")
```
Comment (ConfusedSatlan) [inline src/main/java/com/parkhomovsky/bookstore/model/Book.java]: not right naming, name of column in db is writing with lowercase letters, also look at the task and naming for columns of each variable in db

**Review by ConfusedSatlan — CHANGES_REQUESTED** [review]

File: src/main/java/com/parkhomovsky/bookstore/model/Book.java (hunk 3 above)
Comment (ConfusedSatlan) [reply src/main/java/com/parkhomovsky/bookstore/model/Book.java]: for coverImage it is "cover_image" right name for db column for title it is "title"

Comment (ConfusedSatlan) [reply src/main/java/com/parkhomovsky/bookstore/model/Book.java]: https://medium.com/@centizennationwide/mysql-naming-conventions-e3a6f6219efe

**Review by ConfusedSatlan — APPROVED** [review]

File: src/main/java/com/parkhomovsky/bookstore/BookheroApplication.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    @Autowired
+    private BookService bookService;
+
+    public static void main(String[] args) {
+        SpringApplication.run(BookheroApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
```
Comment (vovasalyha) [inline src/main/java/com/parkhomovsky/bookstore/BookheroApplication.java:23]: use lambda

File: src/main/java/com/parkhomovsky/bookstore/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
```
Comment (vovasalyha) [inline src/main/java/com/parkhomovsky/bookstore/model/Book.java:18]: It's not necessary to add name for each column, but let it be.

File: src/main/java/com/parkhomovsky/bookstore/repository/implementation/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> allBooksQuery = session.createQuery("from Book", Book.class);
```
Comment (vovasalyha) [inline src/main/java/com/parkhomovsky/bookstore/repository/implementation/BookRepositoryImpl.java:47]:
```suggestion
            Query<Book> allBooksQuery = session.createQuery("FROM Book", Book.class);
```

**Review by vovasalyha — APPROVED** [review]
GJ. Please take a look on few suggestions.

**Pull request merged**